spec/Appendix C -- Schema Coordinates.md

```diff
@@ -0,0 +1,93 @@
+# Appendix: Schema Coordinates
+
+Schema Coordinates are human readable strings that uniquely identify an element defined in a GraphQL Schema.
+
+## Definition 
+
+SchemaCoordinates :
+  - TypeDefinitionName FieldSpecifier?
+  - EnumName EnumValueSpecifier?
+  - @ DirectiveName ArgumentSpecifier?
+  - UnionName
+
+TypeDefinitionName:
+  - ObjectTypeName
+  - InterfaceTypeName
+
+FieldSpecifier :
+  - . FieldName ArgumentSpecifier?
+
+ArgumentSpecifier :
+  - ( ArgumentName: )
+
+EnumValueSpecifier :
+  - . EnumValue
+
```

Comment:
Also, it would be great to have a defined semantics to resolve something from a schema coordinate.

An example of this in the spec already are string semantics - https://spec.graphql.org/draft/#sec-String-Value.Semantics

This would avoid needing to completely rely on examples to illustrate all possible use cases

Reply:
@magicmark and @benjie - folding your ideas together and adding semantics, how do you feel about this:

```
SchemaCoordinate:
  - Name
  - Name . Name
  - Name . Name ( Name : )
  - @ Name
  - @ Name ( Name : )

**Semantics**

SchemaCoordinate: Name
  1. Let {typeName} be the value of the first {Name}.
  2. Return the type in the schema named {typeName}.

SchemaCoordinate: Name . Name
  1. Let {typeName} be the value of the first {Name}.
  2. Let {type} be the type in the schema named {typeName}.
  3. Assert: {type} must exist.
  4. If {type} is an Enum type:
    1. Let {enumName} be the value of the second {Name}.
    2. Return the enum value of {type} named {enumName}.
  5. Otherwise if {type} is an Input Object type:
    1. Let {inputFieldName} be the value of the second {Name}.
    2. Return the input field of {type} named {inputFieldName}.
  6. Otherwise {type} must be an Object or Interface type:
    1. Let {fieldName} be the value of the second {Name}.
    2. Return the field of {type} named {fieldName}.
  
SchemaCoordinate: Name . Name ( Name : )
  1. Let {typeName} be the value of the first {Name}.
  2. Let {type} be the type in the schema named {typeName}.
  3. Assert: {type} must be an Object or Interface type.
  4. Let {fieldName} be the value of the second {Name}.
  5. Let {field} be the field of {type} named {fieldName}.
  6. Assert: {field} must exist.
  7. Let {argumentName} be the value of the third {Name}.
  8. Return the argument of {field} named {argumentName}.

SchemaCoordinate: @ Name
  1. Let {directiveName} be the value of the first {Name}.
  2. Return the directive in the schema named {directiveName}.

SchemaCoordinate: @ Name ( Name : )
  1. Let {directiveName} be the value of the first {Name}.
  2. Let {directive} be the directive in the schema named {directiveName}.
  3. Assert: {directive} must exist.
  7. Let {argumentName} be the value of the second {Name}.
  8. Return the argument of {directive} named {argumentName}.
```

Reply:
>   5. Otherwise {type} must be an Object or Interface type:

Maybe 
> 5. Otherwise:
>   1. Assert {type} is an Object or Interface type.
>   2. Let {fieldName} be...

LGTM :+1:

Reply:
wow!

Ok this totally makes sense. I think I understand how to write the grammar a bit better now - I was trying to define the tokens by awkwardly encoding it with the naming of the token itself (e.g. "InterfaceTypeName").

This + the semantics section is clearly better!

Thank you so much @benjie and @leebyron for your work on this! <3